### PR TITLE
docs: set up linkinator for doc site

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/envoyproxy/gateway/issues/2979

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -31,6 +31,12 @@ jobs:
...
 
+    - name: Install linkinator
+      run: npm install -g linkinator
+
+    - name: Check links
+      run: make docs docs-check-links
+
   docs-build:
     runs-on: ubuntu-latest
     needs: docs-lint
```
Comment (arkodg) [inline .github/workflows/docs.yaml:42]:
```suggestion
    needs: docs-lint
```

File: tools/make/docs.mk [hunk 2]
```diff
@@ -81,6 +81,11 @@ docs-release-gen:
 	@echo '  version = "$(TAG)"' >> site/hugo.toml
 	@echo '  url = "/$(TAG)"' >> site/hugo.toml
 
+.PHONY: docs-check-links
+docs-check-links:
+	@$(LOG_TARGET)
+	linkinator site/public/ -r --concurrency 25 -s "localhost:3000 localhost:9090 k8s.io ntia.gov github.com example.com _print latest v0.6.0 v0.5.0 v0.4.0 v0.3.0 v0.2.0 v0.1.0"
```
Comment (arkodg) [inline tools/make/docs.mk:87]: can you add a comment in here about what the arg `-s` does ?

File: .github/workflows/docs.yaml [hunk 3]
```diff
@@ -31,6 +31,12 @@ jobs:
         files: site/content/*
         config_file: ".github/markdown_lint_config.json"
 
+    - name: Install linkinator
+      run: npm install -g linkinator
+
+    - name: Check links
+      run: make docs docs-check-links
```
Comment (arkodg) [inline .github/workflows/docs.yaml:38]:
```suggestion
      run: make docs-check-links
```

**Comment by arkodg** [comment]
thanks for fixing the dead links and automating the detection of it 🙈

File: .github/workflows/docs.yaml (hunk 3 above)
Comment (zirain) [reply .github/workflows/docs.yaml:38]: why remove `docs`? without it there's nothing under `site/public`, or what I missed?

File: site/content/en/v1.0.0/user/traffic/udp-routing.md [hunk 4]
```diff
@@ -1,24 +1,18 @@
...
-__Note:__ UDPRoute allows Envoy Gateway to operate as a non-transparent proxy between a UDP client and server. The lack
-of transparency means that the upstream server will see the source IP and port of the Gateway instead of the client.
-For additional information, refer to Envoy's [UDP proxy documentation][].
+**Note:** UDPRoute allows Envoy Gateway to operate as a non-transparent proxy between a UDP client and server. The lack of transparency means that the upstream server will see the source IP and port of the Gateway instead of the client. For additional information, refer to Envoy's [UDP proxy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/udp_filters/udp_proxy).
 
 ## Prerequisites
 
-Follow the steps from the [Quickstart](../../quickstart) guide to install Envoy Gateway and the example manifest.
-Before proceeding, you should be able to query the example backend using HTTP.
+Follow the steps from the [Quickstart](../quickstart) guide to install Envoy Gateway and the example manifest. Before proceeding, you should be able to query the example backend using HTTP.
```
Comment (Xunzhuo) [inline site/content/en/v1.0.0/user/traffic/udp-routing.md:11]: this is  not right I think, quickstart is outside of traffic dir

Comment (Xunzhuo) [reply site/content/en/v1.0.0/user/traffic/udp-routing.md:11]: Checking it manually is hard to maintain, do not wish to bring more broken links

Comment (zirain) [reply site/content/en/v1.0.0/user/traffic/udp-routing.md:11]: that's why istio/eg use linkinator, it will help at the most of cases.

**Pull request closed without merging**